src/pages/aidataset/[accession_id].astro

```diff
@@ -20,6 +22,9 @@ const { accession_id } = Astro.params;
 const study = exports["datasets"][accession_id];
 
 const images = study.image_uuids.map((uuid) => exports["images"][uuid]);
+
+const viewable_non_annotation_images = (study.image_uuids.filter(item => !Object.values(study.annotation_images).includes(item))).map((uuid) => exports["images"][uuid]);
+const viewable_annotation_images = (study.image_uuids.filter(item => Object.values(study.annotation_images).includes(item))).map((uuid) => exports["images"][uuid]);
```

Comment:
Replacing Object.values(study.annotation_images).includes(item) with study.annotation_images[item] doesn't work as the item is value not key in the study.annotation_images dictionary.

Reply:
Is there any other way to split this up and make it more readable?

Reply:
I couldn't find a way; feel free to change it if you do. Alternatively I can add a comment that explains what the code does.

Reply:
You're right, I thought annotation_images one of those objects indexed by uuid (like images) - which sort of proves the readability point

Are these types of queries on the export used in multiple places? Maybe helpers for data export manipulation (even if they're one-liners) would be useful - just because they're named so they say what they do for free. Could just be a file with a bunch of functions until it becomes unwieldy.

Also (this is probably out of scope though) naming conventions help a lot with understanding what an attribute is. E.g. image_uuids says that it's a bunch of uuids. Images to me suggests a list of images, whereas images_by_uuid suggests a map. Same for annotation_images, I'm guessing the key is an annotation the value is an image, but I only see that they're uuids (instead of full objects I guess) if I click through to the definition. If it were annotation_uuid_to_image_uuid (or flipped with by instead of to / or something shorter, just with regular structure) it would tell me what kinds of things it contains

Reply:
I don't think this is used elsewhere. I can take this function into the bia_exporter instead of here and just pass more lists which would make things simpler here but then it has the issue of passing the same information in various different ways. Also, all of this will probably change once we implement AnnotationFile and related classes into the models.

Reply:
Having only helpers touch the raw data (the export in this case) is even more useful if you plan to refactor since it makes it easier to tell what needs to change. Are there any other options to address data extraction generally being hard to read?